src/Orleans.Core/Messaging/CachingSiloAddressCodec.cs

```diff
@@ -169,37 +170,21 @@ private static SiloAddress ReadSiloAddressInner<TInput>(ref Reader<TInput> reade
         private static void WriteSiloAddressInner<TBufferWriter>(ref Writer<TBufferWriter> writer, SiloAddress value) where TBufferWriter : IBufferWriter<byte>
         {
             var ep = value.Endpoint;
-            Span<byte> buffer = stackalloc byte[64];
-            if (ep.Address.TryWriteBytes(buffer, out var length))
-            {
-                var writable = writer.WritableSpan;
-                if (writable.Length > length)
-                {
-                    // IP
-                    writer.WriteVarInt32(length);
-                    buffer.Slice(0, length).CopyTo(writable[1..]);
-                    writer.AdvanceSpan(length);
-
-                    // Port
-                    writer.WriteVarUInt32((uint)ep.Port);
-
-                    // Generation
-                    writer.WriteInt32(value.Generation);
-
-                    return;
-                }
-            }
+            Unsafe.SkipInit(out Guid tmp);
+            var buffer = MemoryMarshal.AsBytes(MemoryMarshal.CreateSpan(ref tmp, 1));
+            ep.Address.TryWriteBytes(buffer, out var length);
+            Debug.Assert(length > 0);
```

Comment:
As with the other comment, I think if we're going to rely on IPAddress never exceeding 16 bytes, then we should make this a non-conditionally compiled check.